helios-testing/src/main/java/com/spotify/helios/testing/HeliosSoloDeployment.java

```diff
@@ -103,12 +115,20 @@
   private final HeliosClient heliosClient;
   private boolean removeHeliosSoloContainerOnExit;
   private final int jobUndeployWaitSeconds;
+  private final SoloMasterProber soloMasterProber;
+  private final SoloHostProber soloHostProber;
+
+  private static volatile HeliosDeployment soloDeployment;
```

Comment:
this field is only used within the`getOrCreateHeliosSoloDeployment()` method; I think it would be cleaner if the reference to the static HeliosSoloDeployment was moved to TemporaryJobs rather than the HeliosSoloDeployment class itself.

In other words - make the class that wants to use HSD and doesn't care if it gets an old instance or creates a new instance be the thing that holds the reference.